### PR TITLE
various changes to increase resiliency when network is down

### PR DESCRIPTION
**Pull request by mikz**
do not crash, rather try to handle errors gracefully

- network down
- missing nameservers
- ipv6
- html error messages

### CODE REVIEW

File: apicast/src/provider.lua [hunk 1]
```diff
@@ -39,20 +39,32 @@ local _M = {
...
 
   _M.contents = configuration.encode(contents)
-  _M.configured = true
-  _M.configuration = config
-  _M.services = config.services or {} -- for compatibility reasons
+
+  if config then
+    _M.configured = true
+    _M.configuration = config
+    _M.services = config.services or {} -- for compatibility reasons
```
Comment (davidor) [inline apicast/src/provider.lua]: Would be good to know what those compatibility reasons are ... :)

File: apicast/src/configuration.lua [hunk 2]
```diff
@@ -435,7 +439,8 @@ function _M.download(endpoint)
   if body and res.status == 200 then
     ngx.log(ngx.DEBUG, 'configuration response received:' .. body)
 
-    local ok, err = _M.decode(body)
+    local ok
```
Comment (mikz) [inline apicast/src/configuration.lua]: luacheck complaining about shadowing `err`

File: apicast/src/provider.lua (hunk 1 above)
Comment (mikz) [inline apicast/src/provider.lua]: yeah, I don't really know :) I guess when testing with system or for liveness/readiness probes of openshift.

File: apicast/src/configuration.lua (hunk 2 above)
Comment (davidor) [inline apicast/src/configuration.lua]: I see. If we had the code more divided into methods or used more explicit var names this wouldn't be needed, but I'm OK with that for now.

Comment (mikz) [inline apicast/src/configuration.lua]: Yeah. I'm not sure how to solve this. If you have several possible error responses in one method, should they have different names? Or reuse them? Guess more reading of official packages needed :)

**Pull request merged**